src/finalizer/finalizer.cpp

```diff
@@ -38,6 +38,31 @@ extern "C" HRESULT Initialize(int argc, wchar_t* argv[])
     return hr;
 }
 
+extern "C" HRESULT DeleteWorkloadRecords(LPWSTR sczSdkFeatureBandVersion, LPWSTR sczArchitecture)
+{
+    HRESULT hr = S_OK;
+    LPWSTR sczWorkloadRecordsRegistryKeyName = NULL;
+    HKEY hkWorkloadRecordsKey = NULL;
+
+    // MSI based workload records are tied to the underlying SDK host architecture and feature band. Since the feature band is being removed, we 
+    // can remove the records too.
+    hr = StrAllocConcatFormatted(&sczWorkloadRecordsRegistryKeyName, L"SOFTWARE\\Microsoft\\dotnet\\InstalledWorkloads\\Standalone\\%ls", sczArchitecture);
+    ExitOnFailure(hr, "Failed to allocate string for workload records registry path.");
+
+    hr = RegOpen(HKEY_LOCAL_MACHINE, sczWorkloadRecordsRegistryKeyName, KEY_READ | KEY_WRITE, &hkWorkloadRecordsKey);
+    ExitOnFailure(hr, "Failed to open workload records key.");
+    
+    hr = RegDelete(hkWorkloadRecordsKey, sczSdkFeatureBandVersion, REG_KEY_DEFAULT, TRUE);
+    ExitOnFailure(hr, "Failed to delete workload records key.");
+    
+    LogStringLine(REPORT_STANDARD, "Deleted workload records for %ls under %ls.", sczSdkFeatureBandVersion, sczArchitecture);
```

Comment:
This one I'm still thinking about. We're planning an adding something to track workloads from VS, though I guess we could just look at the subkeys and if there aren't any, just delete it completely. Sorry, I'm thinking out load as I'm typing this reply. Okay, let me push another fix for this.

Reply:
Were you still planning to make this change?  I thought of this because it is the MSI behavior for keys and directories.  When MSI removes the last value in a key or file in a directory it will delete that key / directory.  Maybe test it to see what happens here.  I can't recall if dutil!RegDelete does this for you or not.

Reply:
Yes, I'm just working out a good way to traverse everything backwards.

Reply:
So it looks like I need to walk the path backwards, but I can't find a good method in PathUtil or StrUtil to strip off trailing slashes from paths. There's `PathCchRemoveBackslash` but that is only support on Windows8 and later I think, and Server 2012.